pkg/manager/manager.go

```diff
@@ -91,6 +90,50 @@ func (m *SupportBundleManager) getBundlefilesize() (int64, error) {
 }
 
 func (m *SupportBundleManager) Run() error {
+	phases := []struct {
+		Name string
+		Run  func() error
+	}{
+		{
+			PhaseInit,
+			m.phaseInit,
+		},
+		{
+			PhaseClusterBundle,
+			m.phaseCollectClusterBundle,
+		},
+		{
+			PhaseNodeBundle,
+			m.phaseCollectNodeBundles,
+		},
+		{
+			PhasePackaging,
+			m.phasePackaging,
+		},
+		{
+			PhaseDone,
+			m.phaseDone,
+		},
+	}
+
+	for i, phase := range phases {
+		logrus.Infof("running phase %s", phase.Name)
+		m.status.SetPhase(phase.Name)
+		if err := phase.Run(); err != nil {
+			m.status.SetError(err.Error())
+			logrus.Errorf("fail to run phase %s: %s", phase.Name, err.Error())
+			break
+		}
+
+		progress := 100 * (i + 1) / len(phases)
+		m.status.SetProgress(progress)
+		logrus.Infof("succeed to run phase %s. Progress (%d).", phase.Name, progress)
+	}
+
+	select {}
```

Comment:
optional: prefer to use a context single to manage the process, e.g using the `SetupSignalContext()` from the `k8s.io/apiserver/pkg/server`.

Reply:
Setup a context in https://github.com/harvester/support-bundle-utils/pull/6/commits/debbb6926df5895cee7e945d8c62ec854bf7fd1a